src/KnitPlugin.kt

```diff
@@ -52,6 +50,17 @@ class KnitPlugin : Plugin<Project> {
     }
 }
 
+private fun Project.checkDependsOn(other: TaskProvider<*>) {
+    afterEvaluate {
+        // afterEvaluate because the order of plugin application does not reflect declaration order in some Gradle versions
```

Comment:
could you please specify which versions are affected?

Reply:
`afterEvaluate` is unnecessary because you can use `withPlugin`.

Reply:
Thanks for pointing this out, much better.